File: C#/Behavioral/Mediator/DesignPatterns.BryanHansen.Mediator/Light.cs
using System;

namespace DesignPatterns.BryanHansen.Mediator
{
    public class Light
    {
        private bool _isOn;
        private readonly string _location;

        public Light(string location)
        {
            _location = location;
        }

        public void On()
        {
            _isOn = true;
            Console.WriteLine($"{_location} light switched on.");
        }

        public void Off()
        {
            _isOn = false;
            Console.WriteLine($"{_location} light switched off.");
        }

        public void Toggle()
        {
            if (_isOn)
            {
                Off();
            }
            else
            {
                On();
            }
        }
    }
}

File: Java/src/main/java/behavioral/mediator/Light.java
package behavioral.mediator;

// receiver
public class Light {

    private final String name;
    private boolean isOn = false;

    public Light(String name) {
        this.name = name;
    }

    public void toggle() {
        if(isOn) {
            off();
            isOn = false;
        }
        else {
            on();
            isOn = true;
        }
    }

    public boolean isOn() {
        return isOn;
    }

    public void on() {
        System.out.println(name + " light switched on.");
    }

    public void off() {
        System.out.println(name + " light switched off.");
    }
}
